### PR TITLE
41 analog clock widget

### PR DESCRIPTION
**Pull request by isigmund**
These changes add a new widget to display an analog or digital clock on a panel

### CODE REVIEW

File: web/app/widgets/clock/clock.settings.tpl.html.save [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<div>
```
Comment (ghys) [inline web/app/widgets/clock/clock.settings.tpl.html.save]: please remove this file

File: web/app/widgets/clock/clock.widget.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        $scope.items = OHService.getItems();
+
+        $scope.form = {
+            name: widget.name,
+            sizeX: widget.sizeX,
+            sizeY: widget.sizeY,
+            col: widget.col,
+            row: widget.row,
+            mode: widget.mode,
+	    analog_theme: widget.analog_theme,
```
Comment (ghys) [inline web/app/widgets/clock/clock.widget.js]: Please align this :) nobackground also

File: web/gulpfile.js [hunk 3]
```diff
@@ -106,12 +106,16 @@ gulp.task('vendor-js', function() {
         'bower_components/ng-knob/dist/ng-knob.min.js',
         'bower_components/inobounce/inobounce.min.js',
         'bower_components/oclazyload/dist/ocLazyLoad.min.js',
+        'bower_components/angular-ui-clock/dist/angular-clock.min.js',
         'node_modules/n3-charts/build/LineChart.min.js',
-        'vendor/angular-web-colorpicker.js'
+        'vendor/angular-web-colorpicker.js',
```
Comment (ghys) [inline web/gulpfile.js]: the comma is unnecessary

File: web/gulpfile.js [hunk 4]
```diff
@@ -173,4 +177,4 @@ gulp.task('vendor', [
     'vendor-fonts'
 ], function () {});
 
-gulp.task('default', ['vendor', 'codemirror'], function () {});
+gulp.task('default', ['vendor', 'codemirror' /*, 'angular-clock' */ ], function () {});
```
Comment (ghys) [inline web/gulpfile.js]: please remove the commented out angular-clock task

File: web/vendor/vendor.scss [hunk 5]
```diff
@@ -14,4 +14,13 @@
 
 @import "../bower_components/angular-gridster/dist/angular-gridster";
 @import "../bower_components/angularjs-slider/dist/rzslider";
+@import "../bower_components/angular-ui-clock/dist/angular-clock";
 @import "node_modules/n3-charts/build/LineChart";
+
+
+// overwrite font settings for angular clock to use standard roboto font at 100%
```
Comment (ghys) [inline web/vendor/vendor.scss]: can you move this to your clock.scss file, if possible?

**Comment by ghys** [comment]
Great PR!
I'll need you to:
- rebase your work off the latest changes and fix the conflict in index.html (the file had tabs, I changed them to spaces);
- fix a few things as described above;
- sign off all your commits with your real name and email as indicated in the contributing rules (`git commit -s`), and ideally [squash them](https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/) into one;
- Add yourself to AUTHORS if you wish.

Thanks ;)

File: web/app/widgets/clock/clock.settings.tpl.html.save (hunk 1 above)
Comment (isigmund) [inline web/app/widgets/clock/clock.settings.tpl.html.save]: done

File: web/app/widgets/clock/clock.widget.js (hunk 2 above)
Comment (isigmund) [inline web/app/widgets/clock/clock.widget.js]: not sure what you mean here, in my file everything looks nicely aligned

File: web/gulpfile.js (hunk 3 above)
Comment (isigmund) [inline web/gulpfile.js]: done

File: web/gulpfile.js (hunk 4 above)
Comment (isigmund) [inline web/gulpfile.js]: done

File: web/vendor/vendor.scss (hunk 5 above)
Comment (isigmund) [inline web/vendor/vendor.scss]: @ghys: Where do you want me to put the clock.scss file ? At the moment I placed it in the vendor flder and included an @import in the vendor.scss. Is that ok ?

Comment (isigmund) [inline web/vendor/vendor.scss]: done

File: web/app/widgets/clock/clock.widget.js (hunk 2 above)
Comment (isigmund) [inline web/app/widgets/clock/clock.widget.js]: done

**Comment by ghys** [comment]
Thanks @isigmund!

**Pull request merged**